### PR TITLE
Ports Basepixels

### PR DESCRIPTION
**Pull request by Tsar-Salat**
This is a recoup of:
- https://github.com/BeeStation/BeeStation-Hornet/pull/6060

Ports:
- https://github.com/Skyrat-SS13/Skyrat-tg/pull/870
- https://github.com/tgstation/tgstation/pull/54652
- https://github.com/tgstation/tgstation/pull/72309

## About The Pull Request

This is a recoup of Kapu's pr from about a year ago. After discussion with Bacon, I got approval to reopen if the player keybind was removed. https://discord.com/channels/427337870844362753/586312879200927767/1075129296043122688

PIXEL SHIFTING:
Pixelshifting exists in code only, and has no keybind.

> ~~Adds pixel shifting into the game, enabling you to smoothly shift around on the tile you're in. With a few changes.
> Movement is now smooth, you do not need to spam the key you wish to move in, you may hold it down and watch yourself smoothly move.~~
> ~~Keybind is currently C.~~
> 
> ~~This PR does not contain a savefile Burn, existing users will not have this key bound by default. A burn will be added when it is fullmerge ready.~~

BASEPIXELS

> This adds two variables (base_pixel_x and base_pixel_y) and refactors a little of the pixel-shifting code to support better persistent pixel-shifts that would otherwise get reset by most animation and standard shifting procs.
> 
> As of now it doesn't do much besides allowing admins to shift mobs to a different center, but the long-term use of this is to be able to move the mobs up to give a better sense of depth perspective.

also adds a helper to set them correctly and get rid of messed up offsets

## Why It's Good For The Game
Basepixels are better code.

Whilst I understand the concerns with pixelshifting, there's some cool stuff that can be done that code even if we decide never to implement the keybind.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

normal lif
appears normally

</details>

## Changelog
:cl: Rkz, Kapu1178, Azarak, Gandalf2k15, Rohesie
code: adds a neutered pixelshifting system. It has no keybind and thus cannot be accessed by players.
code: Added base_pixel vars to atom/movable, replaced most uses of initial(pixel).
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
There is quite literally no purpose to pixel shifting if it isn't player usable. using `animate()` or just setting `pixel_y/pixel_x` is a better option. Base pixels are good though.

File: code/__DEFINES/keybinding.dm [hunk 1]
```diff
@@ -60,6 +60,10 @@
 #define COMSIG_KB_MOB_TARGETBODYGROIN_DOWN "keybinding_mob_targetbodygroin_down"
 #define COMSIG_KB_MOB_TARGETLEFTLEG_DOWN "keybinding_mob_targetleftleg_down"
 #define COMSIG_KB_MOB_PREVENTMOVEMENT_DOWN "keybinding_mob_preventmovement_down"
+/*
+* Player input currently unauthorized by dev. Keeping this here for compatibility reasons
+* #define COMSIG_KB_MOB_PIXELSHIFT_DOWN "keybinding_mob_pixelshift_down"
+*/
```
Comment (PowerfulBacon) [inline code/__DEFINES/keybinding.dm]: Remove this

File: code/datums/keybinding/mob.dm [hunk 2]
```diff
@@ -414,3 +414,31 @@
...
+	M.shifting = TRUE
+	return TRUE
+
+/datum/keybinding/mob/pixel_shift/up(client/user)
+	. = ..()
+	if(.)
+		return
+	var/mob/M = user.mob
+	M.shifting = FALSE
+	return TRUE
```
Comment (PowerfulBacon) [inline code/datums/keybinding/mob.dm]: Remove these if they are unused and will not be used.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
This is fine but theres no point of having pixel shift code inside mob movement if its inaccessible and won't be added. If the ability to lean against walls is going to be added it should be done by dragging your character against the wall. Players won't get a button to move their offset around.

File: code/datums/keybinding/mob.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/datums/keybinding/mob.dm]: removed in a3b1d4f

File: code/__DEFINES/keybinding.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/__DEFINES/keybinding.dm]: removed in a3b1d4f

**Comment by Crossedfall** [comment]
Checks look to have gotten stuck during an outage

**Comment by Tsar-Salat** [comment]
Alrighty, im also including a setter that should make these have consistent mapping behavior

image

**Comment by Tsar-Salat** [comment]
hold on, I need to adjust this for #8587

**Comment by Tsar-Salat** [comment]
cleared it up with item-pickup author, should be good

https://discord.com/channels/427337870844362753/586312879200927767/1115952856500031591

File: code/game/atoms.dm [hunk 3]
```diff
@@ -1641,6 +1646,24 @@
 	if(ai_controller)
 		ai_controller = new ai_controller(src)
 
+///Setter for the "base_pixel_x" var to append behavior related to it's changing
+/atom/proc/set_base_pixel_x(var/new_value)
```
Comment (itsmeow) [inline code/game/atoms.dm:1664]: ver-editing basepixels should also call this proc

File: code/modules/pixelshifting/pixelshift.dm
Comment (itsmeow) [inline code/modules/pixelshifting/pixelshift.dm:1]: just remove this file, it doesn't do anything

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/atoms.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/game/atoms.dm:1664]: added to atoms.dm in 33979f0

File: code/game/atoms.dm [hunk 4]
```diff
@@ -1113,12 +1118,27 @@
...
+
 	if(!GLOB.Debug2)
 		flags_1 |= ADMIN_SPAWNED_1
+
 	. = ..()
+
 	switch(var_name)
 		if(NAMEOF(src, color))
 			add_atom_colour(color, ADMIN_COLOUR_PRIORITY)
+			update_appearance()
```
Comment (itsmeow) [inline code/game/atoms.dm]: why

Comment (Tsar-Salat) [reply code/game/atoms.dm]: mb

**Review by itsmeow — APPROVED** [review]
It indeed adds base pixels. Cool.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**